cmd/kubeadm/app/cmd/defaults.go

```diff
@@ -71,5 +72,13 @@ func setInitDynamicDefaults(cfg *kubeadmapi.MasterConfiguration) error {
 		}
 	}
 
+	if cfg.NodeName == "" {
```

Comment:
This can actually be trimmed down to `cfg.NodeName = node.GetHostname(cfg.NodeName)`